src/java.desktop/share/classes/com/sun/imageio/plugins/jpeg/JPEGImageReader.java

```diff
@@ -902,6 +902,9 @@ public ImageTypeSpecifier getRawImageType(int imageIndex)
     @Override
     public Iterator<ImageTypeSpecifier> getImageTypes(int imageIndex)
         throws IOException {
+        if (imageIndex != 0) {
+            throw new IndexOutOfBoundsException("imageIndex != 0!");
+        }
```

Comment:
There are JPEG streams with > 1.
And elsewhere we handle it. Look at hasNextImage()

Is this method the sole place where -1 would not throw the right exception ?

Reply:
Thanks for the inputs.

getWidth(), getHeight(), getRawImageType() also need similar check.
What i have noticed is for all negative indices apart from "-1" we end up in gotoImage() function and it throws appropriate exception. Only getWidth(), getHeight(), getRawImageType() & getImageTypes() we don't throw proper exception according to spec when "-1" index is used. Also in case of getRawImageType() we actually throw NPE when "-1" index is passed.

So i have updated the code to add additional check for negative index at these places also.